lumibot/data_sources/ccxt_data.py

```diff
@@ -138,14 +139,22 @@ def get_barset_from_api(self, api, symbol, freq, limit=None, end=None):
         if end is None:
             end = datetime.datetime.utcnow()
 
-        endunix = self.api.parse8601(end.strftime("%Y-%m-%d %H:%M:%S"))
         buffer = 10  # A few extra datapoints in the download then trim the df.
         if freq == "1m":
+            dt = end.replace(second=59, microsecond=999999)
+            end = dt - datetime.timedelta(minutes=1)
             start = end - datetime.timedelta(minutes=limit + buffer)
         else:
+            dt = end.replace(hour=23, minute=59, second=59, microsecond=999999)
+            end = dt - datetime.timedelta(days=1)
```

Comment:
why make these changes? can you please add comments?

Reply:
Sorry about that. It's my mistake, I modified it to download backtest data and it's a problem when downloading real data. I'll change it back.